### PR TITLE
ignore dotfiles and backup files in the watch list

### PR DESCRIPTION
**Pull request by s1037989**

### CODE REVIEW

File: script/prowess [hunk 1]
```diff
@@ -25,7 +25,9 @@ sub parse_argv {
 }
 
 sub run_prove {
-  my ($self, $args) = @_;
+  my ($self, $changed, $args) = @_;
+  do { warn "[prowess] no desired changes\n" if DEBUG; } and return unless $changed;
+  return if DEBUG > 1;
```
Comment (jhthorsen) [inline script/prowess]: I don't think `$changed` should be passed into this method... se my comment later.

File: script/prowess [hunk 2]
```diff
@@ -40,17 +42,20 @@ sub run {
   my $self    = shift;
   my $args    = $self->parse_argv(@_);
   my $watcher = Filesys::Notify::Simple->new($args->{watch});
+  my $ignore  = qr{(?:/\.[^/]+$|\.bak$|\.old$|\.swp$|~$)};
```
Comment (jhthorsen) [inline script/prowess:43]: 👍

File: script/prowess [hunk 3]
```diff
@@ -40,17 +42,20 @@ sub run {
...
+  my $changed = 1;
   my $exit    = 0;
 
   while (1) {
-    my $pid = $self->run_prove($args->{prove});
+    my $pid = $self->run_prove($changed, $args->{prove});
```
Comment (jhthorsen) [inline script/prowess]: Could this be this instead:

```
my $pid = $changed ? $self->run_prove($args->{prove}) : 0;
```

File: script/prowess [hunk 4]
```diff
@@ -40,17 +42,20 @@ sub run {
...
     eval {
-# Try to capture:
-# kevent error: Interrupt innerhalb eines Systemaufrufs at /usr/perl5.20.0/lib/site_perl/5.20.0/Filesys/Notify/KQueue.pm line 114.
+      # Try to capture:
+      # kevent error: Interrupt innerhalb eines Systemaufrufs at /usr/perl5.20.0/lib/site_perl/5.20.0/Filesys/Notify/KQueue.pm line 114.
       $watcher->wait(
         sub {
-          return unless my @changed = grep { !-d $_ } map { $_->{path} } @_;
-          warn "[prowess] changed: @changed\n" if DEBUG;
+          return unless $changed = my @changed = grep { !-d $_ && $_ !~ $ignore } map { $_->{path} } @_;
```
Comment (jhthorsen) [inline script/prowess]: Do you need both @changed and $changed?

**Comment by s1037989** [comment]
Good suggestions!  I think this meets all your feedback!

What do you think about the DEBUG level 2 ability?  Too much?  Unnecessary?  Shoot, proly so...  Rather than testing on a big project with lots of tests, test on app-prowess itself!

File: script/prowess [hunk 5]
```diff
@@ -40,16 +40,18 @@ sub run {
   my $self    = shift;
   my $args    = $self->parse_argv(@_);
   my $watcher = Filesys::Notify::Simple->new($args->{watch});
+  my $ignore  = qr{(?:/\.[^/]+$|\.bak$|\.old$|\.swp$|~$)};
+  my @changed = (1);
   my $exit    = 0;
 
   while (1) {
-    my $pid = $self->run_prove($args->{prove});
+    my $pid = @changed ? $self->run_prove($args->{prove}) : 0;
```
Comment (s1037989) [inline script/prowess:48]: This was way more simple.  Why do sometimes the most simple and obvious solutions elude me??

File: script/prowess [hunk 6]
```diff
@@ -40,16 +40,18 @@ sub run {
...
     eval {
-# Try to capture:
-# kevent error: Interrupt innerhalb eines Systemaufrufs at /usr/perl5.20.0/lib/site_perl/5.20.0/Filesys/Notify/KQueue.pm line 114.
+      # Try to capture:
+      # kevent error: Interrupt innerhalb eines Systemaufrufs at /usr/perl5.20.0/lib/site_perl/5.20.0/Filesys/Notify/KQueue.pm line 114.
       $watcher->wait(
         sub {
-          return unless my @changed = grep { !-d $_ } map { $_->{path} } @_;
+          return unless @changed = grep { !-d $_ && $_ !~ $ignore } map { $_->{path} } @_;
           warn "[prowess] changed: @changed\n" if DEBUG;
```
Comment (s1037989) [inline script/prowess:55]: There was no real value to showing the number of files changed.

**Comment by jhthorsen** [comment]
LGTM!

**Pull request merged**